src/app/carbon-estimation/carbon-estimation.component.ts

```diff
@@ -1,125 +1,79 @@
-
-import { startCase } from 'lodash-es';
-import {
-  EmissionsColours,
-  EmissionsLabels,
-  SVG,
-  getBaseChartOptions,
-  estimatorHeights,
-  tooltipFormatter,
-  placeholderData,
-  ApexChartSeriesItem,
-  ApexChartDataItem,
-} from './carbon-estimation.constants';
+import { ChangeDetectorRef, Component, ElementRef, input, OnDestroy, OnInit, ViewChild } from '@angular/core';
 import { ExpansionPanelComponent } from '../expansion-panel/expansion-panel.component';
-import { Subscription, debounceTime, fromEvent } from 'rxjs';
+import { TabsComponent } from '../tab/tabs/tabs.component';
+import { TabItemComponent } from '../tab/tab-item/tab-item.component';
+import { CarbonEstimationTreemapComponent } from '../carbon-estimation-treemap/carbon-estimation-treemap.component';
+import { CarbonEstimation } from '../types/carbon-estimator';
+import { sumValues } from '../utils/number-object';
+import { estimatorHeights } from './carbon-estimation.constants';
+import { debounceTime, fromEvent, Subscription } from 'rxjs';
+import { CarbonEstimationTableComponent } from '../carbon-estimation-table/carbon-estimation-table.component';
+import { toObservable } from '@angular/core/rxjs-interop';
 
 @Component({
   selector: 'carbon-estimation',
   standalone: true,
-  imports: [NgApexchartsModule, ExpansionPanelComponent],
+  imports: [
+    ExpansionPanelComponent,
+    TabsComponent,
+    TabItemComponent,
+    CarbonEstimationTreemapComponent,
+    CarbonEstimationTableComponent,
+  ],
   templateUrl: './carbon-estimation.component.html',
   styleUrls: ['./carbon-estimation.component.css'],
 })
 export class CarbonEstimationComponent implements OnInit, OnDestroy {
   public carbonEstimation = input<CarbonEstimation>();
   public extraHeight = input<string>();
 
-  public chartData = computed(() => this.getChartData(this.carbonEstimation()));
-  public emissionAriaLabel = computed(() => this.getEmissionAriaLabel(this.chartData(), !this.carbonEstimation()));
+  @ViewChild('detailsPanel', { static: true, read: ElementRef }) detailsPanel!: ElementRef;
+  @ViewChild('treemap', { static: true }) treemap!: CarbonEstimationTreemapComponent;
+
+  public chartHeight!: number;
 
-  public chartOptions = computed(() => this.getChartOptions(!this.carbonEstimation()));
-  private tooltipFormatter = tooltipFormatter;
   private estimatorBaseHeight = sumValues(estimatorHeights);
-
   private resizeSubscription!: Subscription;
+  private hasResized = true;
+  private hasEstimationUpdated = false;
+  private carbonEstimationSubscription?: Subscription;
 
-  @ViewChild('chart') chart: ChartComponent | undefined;
-  @ViewChild('detailsPanel', { static: true, read: ElementRef }) detailsPanel!: ElementRef;
-
-  constructor(private changeDetectorRef: ChangeDetectorRef) {}
+  constructor(private changeDetectorRef: ChangeDetectorRef) {
+    this.carbonEstimationSubscription = toObservable(this.carbonEstimation).subscribe(() => {
```

Comment:
I don't know if this would this be simpler with an `effect()` but it might feel a bit odd calling the signal just for the purposes of receiving updates.

```typescript
effect(() => {
  this.carbonEstimation();
  this.hasEstimationUpdated = true;
});
```

Reply:
I do think it is cleaner than Observables, dont have to deal with the destroying the observable. Yeah slightly annoying you cant define the signal you want it to depend on.